packages/framer-motion/src/value/use-velocity.ts

```diff
@@ -15,8 +16,16 @@ import { useMotionValue } from "./use-motion-value"
 export function useVelocity(value: MotionValue<number>): MotionValue<number> {
     const velocity = useMotionValue(value.getVelocity())
 
-    useMotionValueEvent(value, "velocityChange", (newVelocity) => {
-        velocity.set(newVelocity)
+    const updateVelocity = () => {
+        const latest = value.getVelocity()
+        velocity.set(latest)
+
+        if (latest) frame.update(updateVelocity)
```

Comment:
This confused me for a second, but I think i understand it now. Am I right that this is intentionally creating a loop so that if the value stops changing, the velocity continues to be updated until its reported as 0?

Reply:
Yeah that's right, let me add a comment